tool/internal/setup/find.go

```diff
@@ -30,51 +32,83 @@ func (d *Dependency) String() string {
 	return fmt.Sprintf("{%s@%s: %v}", d.ImportPath, d.Version, d.Sources)
 }
 
-// findCompileCommands finds the compile commands from the build plan log.
-func findCompileCommands(buildPlanLog *os.File) ([]string, error) {
-	const buildPlanBufSize = 10 * 1024 * 1024 // 10MB buffer size
+// parseCdDir extracts the directory path from a "cd" command line (case-insensitive).
+func parseCdDir(line string) (string, bool) {
+	if !strings.HasPrefix(strings.ToLower(line), "cd ") {
+		return "", false
+	}
+	const cdCommandSplitLimit = 2 // Split "cd dir" into [dir, rest] to ignore trailing comments
+	parts := strings.SplitN(line[3:], " ", cdCommandSplitLimit)
+	return strings.TrimSpace(parts[0]), true
+}
+
+// isCgoCommand checks if the line is a cgo tool invocation with -objdir and -importpath flags.
+func isCgoCommand(line string) bool {
+	return strings.Contains(line, "cgo") &&
+		strings.Contains(line, "-objdir") &&
+		strings.Contains(line, "-importpath") &&
+		!strings.Contains(line, "-dynimport")
+}
 
-	// Filter compile commands from build plan log
-	compileCmds := make([]string, 0)
-	scanner := bufio.NewScanner(buildPlanLog)
-	// Seek to the beginning of the file before reading
-	_, err := buildPlanLog.Seek(0, 0)
+// parseBuildPlan scans the build plan log once and extracts both compile commands
+// and CGO object directory mappings.
+func parseBuildPlan(buildPlanLog *os.File) ([]string, map[string]string, error) {
```

Comment:
I think it would be better to introduce a struct type for this.
```suggestion
type buildPlan struct {
   TBD []string
   TBD map[string]string
}
func parseBuildPlan(buildPlanLog *os.File) (buildPlan, error) {
```